db_facts/aws_secrets_manager.py

```diff
@@ -0,0 +1,31 @@
+import json
+
+import boto3
+
+from .db_facts_types import AWSSecret, AWSSecretUsernamePassword
+from .db_type import canonicalize_db_type, db_protocol
+
+def pull_aws_secrets_manager_secret(sm_entry_name: str) -> AWSSecret:
+    response = boto3.client('secretsmanager').get_secret_value(SecretId=sm_entry_name)
+    return json.loads(response['SecretString'])
+
+def pull_aws_secrets_manager_username_password(sm_entry_name: str) -> AWSSecretUsernamePassword:
+    secret = pull_aws_secrets_manager_secret(sm_entry_name)
+    return {
+        'user': secret['username'],
+        'password': secret['password']
+    }
+
+def db_info_from_secrets_manager(sm_entry_name: str):
+    response = pull_aws_secrets_manager_secret(sm_entry_name)
+
+    result = {key.lower(): value for key, value in response.items()}
+    
+    result['host'] = result.pop('hostname')
+    result['user'] = result.pop('username')
+
+    if result.get('type'):
```

Comment:
This is both checking that the key exists and that the associated value is considered True.  Might be better to check for the key explicitly so we can handle those two cases differently.  For example, if type is an empty string vs type is not defined.

Reply:
If type is an empty string, the clause that gets executed in this if statement will just create a new key value pair in the dict {"protocol": ""}, which doesn't seem totally useful but actually would be necessary to bring it in alignment with the behavior of the lpass.py module. I'm gonna just get rid of the conditionality, although we might want to issue a warning or something if type isn't defined.

Reply:
Actually I'm going to leave the if block for now, because of an incompatibility with mypy and `result.get('type')` due to the use of an Optional[Any] type.

See the new code here https://github.com/bluelabsio/db-facts/blob/secrets_manager/db_facts/aws_secrets_manager.py#L29

Reply:
Looks good